### PR TITLE
[SYCL] [NATIVECPU] Integrate OneAPI Construction Kit vectorizer

### PR DESCRIPTION
**Pull request by PietroGhg**
This PR enables Whole Function Vectorization for the Native CPU backend by integrating the OneAPI Construction Kit's vectorizer.

### CODE REVIEW

File: clang/lib/CodeGen/BackendUtil.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include "llvm/CodeGen/RegAllocRegistry.h"
 #include "llvm/CodeGen/SchedulerRegistry.h"
 #include "llvm/CodeGen/TargetSubtargetInfo.h"
+#include "llvm/Frontend/Debug/Options.h"
```
[the remark this answers is not in the document]
Comment (PietroGhg) [reply clang/lib/CodeGen/BackendUtil.cpp]: It is left over from some older changes, I've removed it. Thanks for spotting it

**Review by elizabethandrews — APPROVED** [review]
FE changes LGTM

**Review by uwedolinsky — APPROVED** [review]

**Comment by PietroGhg** [comment]
HI @intel/dpcpp-tools-reviewers, can I get some feedback on this PR?

**Review by smanna12 — APPROVED** [review]
FE changes LGTM

**Comment by PietroGhg** [comment]
@intel/dpcpp-tools-reviewers friendly ping :)

**Review by AlexeySachkov — APPROVED** [review]

**Comment by PietroGhg** [comment]
Hi @intel/llvm-gatekeepers, this looks ready to be merged, thank you

**Comment by sommerlukas** [comment]
Compilation of the project failed in post-commit. I've put up https://github.com/intel/llvm/pull/12852 to temporarily revert this PR while the reason for the failure is investigated.

**Pull request merged**